rfcs/0136-stabilize-incrementally.md

```diff
@@ -0,0 +1,386 @@
+This should build trust between opposing individuals and factions, giving us a foundation upon which to tackle the more challenging material in subsequent steps.
+
+We recognize that Flakes is widely used and will take precautions to ensure users are reasonably informed of any breaking changes that might occur from the stabilization.
+
+# Detailed design
+[design]: #detailed-design
+
+The detailed design of this RFC consists of these parts:
+
+- Establishing Principles
+    - Layering principles
+    - Stabilization process
+
+- Plan bringing Nix into compliance with the principles, and specifying the order in which the outstanding unstable CLI and Flakes features will be tackled.
+
+## Layering principles
+
+These basic layering principles will be added to the [Nix architecture documentation](https://nixos.org/manual/nix/stable/architecture/architecture.html):
+
+### For Nix as as a whole
+
+- **Public Interfaces**
+
+  Layers are not just an implementation detail, instead they are publically exposed to the user via stable interfaces
+  All exposed interfaces in Nix, both for computers or humans, must be matched to layers
+
+### For each layer
+
+- **Clarity of purpose**
+
+  Layers should not be too "thick".
+  Layers should "do one thing, and do it well".
+
+- **Compositionality**
+
+  Layers should stand alone:
+
+  - They should work as the top layer
+  - They should also work *not* as the top layer, and with multiple possible layers above them
+  - They should expose a clear interface, which is what makes the previous point possible.
+
+- **Gravity**
+
+  Features should be in the lowest layer it makes sense to have them.
+
+## The general stabilization process --- audit, refine, and *then* stabilze
+
+See the [current documentation on experimental features and their lifecyle](https://nixos.org/manual/nix/stable/contributing/experimental-features.html).
+
+Stabilization of any feature, not just the CLI or Flakes, is not a matter of just flipping a switch on an implementation that has accrued for a period of time.
+Because the moment before stabilization is our last chance to make major changes, it is crucial that we look over what is being stabilized.
+
+To stabilize a piece of functionality (experimental -> stable in flowchart in linked documentation) we must do these things:
+
+1. **Audit the functionality**
+
+   Make note of the current state.
+   Do this publicly so the Nix community at large has a chance to weigh in.
+
+   Checklist during audit:
+
+   - **Documentation**
+
+     Ideally the feature is already well-documented and the audit brings up nothing new.
+     But if it isn't, it must be by the end of the audit.
+
+   - **Whole feature flag, not part of a feature flag**
+
+     It should be possible to enable just the experimental that is ready for stabilization *in isolation*, without also enabling other unstable functionality that is not ready for stabilization.
```

Comment:
This should say “It should be possible to enable just the experimental **features** that[…]”